crates/sys/src/bindings.rs

```diff
@@ -1936,8 +1937,8 @@ extern "C" {
     pub fn strlcpy(
         arg1: *mut core::ffi::c_char,
         arg2: *const core::ffi::c_char,
-        arg3: usize,
-    ) -> usize;
+        arg3: core::ffi::c_uint,
+    ) -> core::ffi::c_uint;
```

Comment:
Did you generate the bindings using a Windows machine? We've seen these flip-flop between `c_uint` and `usize` several times now as the bindings were updated.

Reply:
Up to this point, yes, but I'm planning to use Linux (or WSL2) for any future updates.

Reply:
As it turns out, this inconsistency is not because of linux vs windows but because of debian (and presumably all its derivatives like ubuntu) vs everybody else. Debian has like a million patches for llvm and one of them looks like this https://salsa.debian.org/pkg-llvm-team/llvm-toolchain/-/blob/14/debian/patches/23-strlcpy_strlcat_warning_removed.diff. Because of this patch debian's version of libclang.so (clang api library that is used by bindgen) does not consider `strlcpy` a builtin function. As a result, it becomes a regular user-defined function, and so its argument types get a `spelling = size_t` property in the clang AST that bindgen is able to parse and generate `usize` for them. 

I feel like saying that one must use a specific platform (debian linux in this case) to generate the bindings because of its quirks is wrong, and we should instead recommend a platform-independent way to do it. In this particuar case we could recommend using the upstream release artifacts, e.g. https://github.com/llvm/llvm-project/releases/tag/llvmorg-16.0.0. That's what I've been using and that's why it flip-flopped to `core::ffi::c_uint` this time. Thoughts on this?

There's another flip-flop incoming that starts with clang 16 (it's due to [this change](https://github.com/llvm/llvm-project/commit/9e956995db1fc7e792e3dfb3a465a52626195557)). It looks like this in the current pr
```diff
+pub type __gnuc_va_list = u32;
+pub type va_list = __gnuc_va_list;
-pub type va_list = u32;
```
Since I used the upstream release of clang 16, I hit both of them here. I can revert this one for now but eventually it will need to stay like this.

Reply:
Great investigation tracking that down! It's a bit annoying, but at least it appears relatively harmless.

My ultimate goal is to eliminate the prebuilt `bindings.rs` (#41) due to the fact that it currently ties us to a specific Flipper Zero SDK version and makes it difficult to build against a dev branch. I'm about halfway there (https://github.com/flipperzero-rs/flipperzero/tree/build-sys), I just need to finish it off and then do a bunch of testing to make sure it works in multiple environments.

So I'm happy to leave it as it is, at least for the time being.

Reply:
Given the above analysis, I've opened #72 to fix up the `Dockerfile` for generating bindings to be consistent with the non-Debian platforms.

Reply:
Much appreciated, but I was going to update the Dockerfile in this pr (or quickly after) and so I did 😅